EDM-213: Added polarion case OCP-75523 A device can upgrade with embe…

Samudelacruz77 opened the pull request:
Added upgrade case to v4 with infinity sleep app that doesn't run

siserafin commented on test/e2e/agent/agent_updates_test.go:
```diff
@@ -26,7 +26,7 @@ var _ = Describe("VM Agent behavior during updates", func() {
 
 	Context("updates", func() {
-		It("should update to the requested image", Label("updates"), func() {
+		It("should update to the requested image", Label("updates", "OCP-75523"), func() {
```
I think that we can't share the entire internal test case id, but just the number (75523), since it is a public repository. (At least this was a rule in other public projects)

mangelajo replied:
I think we may want to add a prefix to those otherwise it's a bit cryptic if somebody from outside redhat comes and sees "75523", what's that supposed to mean.

May be we should use "rh-qe-75523" or "rh-ocp-75523" ? any preference ?

mangelajo commented on test/e2e/agent/agent_updates_test.go:
```diff
@@ -67,6 +67,52 @@ var _ = Describe("VM Agent behavior during updates", func() {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":v4"
```
You need to add the script changes to create :v4 image

mangelajo replied:
What builds :v4 ? we need the script changes to do that :)

Samudelacruz77 replied:
v4 has an embedded app that tries to start a podman container with infinite sleep but fails to do so, its the one recomended by @hexfusion for this test

hexfusion replied:
https://github.com/flightctl/flightctl/blob/main/test/scripts/agent-images/create_agent_images.sh#L8

Samudelacruz77 commented on test/e2e/agent/agent_updates_test.go:
```diff
@@ -67,6 +67,52 @@ var _ = Describe("VM Agent behavior during updates", func() {
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// related to: https://issues.redhat.com/browse/EDM-679
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			logrus.Info("Device updated to new image 🎉")
```
Yes will change

mangelajo commented on test/e2e/agent/agent_updates_test.go:
```diff
@@ -67,6 +67,52 @@ var _ = Describe("VM Agent behavior during updates", func() {
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
```
```suggestion
			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
``` 

This is just a string for logging in the tests, nothing compared to.

Samudelacruz77 commented:
Modified as requested, now upgrade image is shown in the logs, added a new step to go back to base image after updating to v4 and check the embedded application

siserafin commented on test/e2e/agent/agent_updates_test.go:
```diff
@@ -65,7 +66,80 @@ var _ = Describe("VM Agent behavior during updates", func() {
+						conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+
+			logrus.Infof("Device updated to new image %s 🎉", "flightctl-device:base")
```
Nice! Maybe can we add here also a check that the containers of the old image with embedded app are not running and are removed?

mangelajo commented:
https://github.com/flightctl/flightctl/pull/657/commits  something has gone really wrong with the commits, even though the change is just 2 files.

hexfusion commented:
Love this case nice work!

hexfusion commented:
try 
git fetch <name of upstream>
git rebase <name of upstream>/main
git push -f
